docs(root): update select code and guidance to show custom elements (#986)
mi6/ic-design-system: src/content/structured/components/select/guidance.mdx

### Option list

> The options in the option list display a label by default and can also include a description, or custom elements. Any custom elements used should be non-interactive, such as status tags or icons.

Reviewer: ```suggestion
The options display a label by default for the option list, which can also include a description or custom elements. Any custom elements used should be non-interactive, such as status tags or icons.
```
